objects/src/accounts/storage/mod.rs

```diff
@@ -96,7 +104,17 @@ impl AccountStorage {
         let slots = SimpleSmt::<STORAGE_TREE_DEPTH>::with_leaves(entires)
             .map_err(AccountError::DuplicateStorageItems)?;
 
-        Ok(Self { slots, layout })
+        // check if the provided Vec<Smt> is longer than 255 and return an error if so
+        if let Some(ref m) = maps {
```

Comment:
I think the check here should be that the number of `maps` is exactly equal to the number of `StorageSlotType::Map` entries in `items`.